contrib/babelfishpg_tds/src/backend/tds/tdsprotocol.c

```diff
@@ -123,7 +123,7 @@ ResetTDSConnection(void)
 {
 	const char *isolationOld;
 
-	Assert(TdsRequestCtrl->request == NULL);
+	Assert(resetTdsConnectionFlag || TdsRequestCtrl->request == NULL);
```

Comment:
Should we use AND or OR? I assume AND as both of the conditions must be true now.

Reply:
It is rightly an OR condition, please check the use of resetTdsConnectionFlag incase of sp_reset_connection procedure